lib/Dialect/HW/InstanceImplementation.cpp

```diff
@@ -97,7 +97,8 @@ LogicalResult instance_like_impl::verifyInputs(ArrayAttr argNames,
       return failure();
     }
 
-    if (argNames[i] != moduleArgNames[i]) {
+    if (argNames[i] != moduleArgNames[i] &&
```

Comment:
The problem is the comparison is to the wrong thing, not that it might sometimes be empty.  There are non-empty strings which cannot be represented by ssa names which will exhibit the same problem.